sentry/src/main/java/io/sentry/TraceContext.java

```diff
@@ -50,8 +59,39 @@ public final class TraceContext implements JsonUnknown, JsonSerializable {
         new Dsn(sentryOptions.getDsn()).getPublicKey(),
         sentryOptions.getRelease(),
         sentryOptions.getEnvironment(),
-        user != null ? new TraceContextUser(user) : null,
-        transaction.getName());
+        user != null ? user.getId() : null,
+        user != null ? getSegment(user) : null,
+        transaction.getName(),
+        sampleRateToString(sentryOptions.getTracesSampleRate()));
+  }
+
+  private static @Nullable String getSegment(final @NotNull User user) {
+    final Map<String, String> others = user.getOthers();
+    if (others != null) {
+      return others.get("segment");
```

Comment:
Should `segment` be a new field in the `User` [protocol](https://develop.sentry.dev/sdk/event-payloads/user/) maybe?

Reply:
JS seems to have it https://github.com/getsentry/sentry-javascript/blob/master/packages/types/src/user.ts while cocoa does not. Didn't check any other SDKs. Not sure what exactly this is used for / if it should be added. Can create a separate issue to track if you want.

Reply:
Yes, please, but let's be sure that the approach with `others` works if not now.
Also, in this issue, we should upgrade the protocol https://develop.sentry.dev/sdk/event-payloads/user/ (adding the new field).